src/hamiltonian/PersistenceConstraint.js

```diff
@@ -199,8 +205,11 @@ class PersistenceConstraint extends SoftConstraint {
 					for( let j = 0 ; j < dx.length ; j ++ ){
 						dx[j] *= ld
 					}
-					this.celldirections[t] = dx
 				}
+				// regardless of whether PERSIST is used, 
+				//the dx vector should be normalised and added to cell directions
+				this.normalize(dx)
+				this.celldirections[t] = dx
```

Comment:
1) Note that in the for-loop, we multiply the cell's direction vector dx with lambda_dir. This happens here instead of in deltaH for efficiency reasons (see earlier comments); I am adding a comment for transparency reasons. 
This also means that we should _not_ normalize dx at the end (that would be inefficient: first normalize, then scale by lambda, then normalize to unit length again). 

2) As for the cell directions, I am not sure I completely understand what you mean. In the upper for loop above, the new cell direction is computed from:

- the empirical direction `dx` over the last delta_t steps; this has been computed in the upper part of the postMCSListener.
- the current preferred direction of the cell, stored in `this.celldirections[t]`

These two should then be combined depending on the value of `PERSIST` (`per`):
- at persistence 0, cells completely adjust their preferred direction to their recent empirical direction; they have no persistent influence of their previous `celldirection`. That is: `celldirections` completely overwritten by dx.
- at persistence 1, cells should *not* adjust their intrinsic `celldirection` based on empirical direction over the last steps; they keep the same direction. 
- for 0 < per < 1, both the empirical and the intrinsic direction contribute (in weighted fashion) to the new intrinsic direction.  

But with the suggested change, something strange would happen: for per == 1, everything in the if-statement will be skipped; thus, `dx` will still just be the empirical direction, and that will completely overwrite the old `celldirection`. In other words: as`per` goes from 0 -> 0.99999 the empirical direction will gradually weigh less, and then suddenly at `per` = 1 it will be back at full force.

I therefore think this change is incorrect (unless I am missing something) and should be reverted.